### ISSUE TITLE
what's the difference between this repo and the one at github.com/imathis/octopress

### ISSUE THREAD

**Issue by arielelkin** [octopress/octopress#148]

**Comment by after100** [comment]
This 1 comes with extra magic dust on top











- - - 
- - - 
- - - 
- - - 
- - - 
- - - 
- - - 
- - - 
READ THE FUCKING HOMEPAGE.
HERE. I WILL THROW YOU A BONE http://octopress.org/ 
YOU NEED TODO THE REST

**Comment by arielelkin** [comment]
Dude what the *fuck* is your problem?? 

The project's homepage doesn't actually describe the difference between the two repos. 

It's not clear: It that tells you that "Octopress 2.0 surfaces" and that "Octopress 3.0 is coming". It's really a blog and its last post was from January 2015. WTF?

And assuming that the homepage is the source of truth, it still doesn't adhere to software versioning standards: it doesn't tell you *if a release is stable* and *which release is stable* or if one release *is* a beta. 

The readme of https://github.com/imathis/octopress says that "Octopress 3.0 is in development at https://github.com/octopress/octopress". That's like saying "The last version of Microsoft Word is in development at Microsoft". Well thank you very much. *Which one am I supposed to use?* *Can I migrate from 2.0 to 3.0?* Is 3.0 the canonical version? If so, why isn't 2.0 deprecated? 

And why a separate repository?? Why not transfer ownership of https://github.com/imathis/octopress to https://github.com/octopress/octopress ? Work on new versions on a separate git branch? 

Basically, you're gonna confuse people if you don't follow standards, such as Git flow and semantic versioning, and this confusion should be minimised by at least offering some rationale in the README. 

Instead, we're just told it's "Jekyll's Ferrari". Damn right: it's obnoxiously made and bought by jerks. 

@after100 learn the fucking standards. here: [software release cycle](https://en.wikipedia.org/wiki/Software_release_life_cycle), [Git branching](http://nvie.com/posts/a-successful-git-branching-model/), [semantic versioning](http://semver.org/). 

@imathis if you still care, it would be nice if you stepped in and clarified these matters.

**Comment by pahles** [comment]
@arielelkin While the comment of @after100 lacks any kind of subtlety, maybe you should get off your high horse in telling people how things have to be done. The world isn't that ideal. Using the word "fuck" several times won't help either.

**Comment by arielelkin** [comment]
Let me tell you what doesn't help:

* People who intervene in discussions without actually addressing the matter at hand
* People who get scandalised at expletives
* People who think that "the world isn't that ideal" is an excuse for mediocrity and bad manners
* People who think that being on a high horse is using *GitHub fucking Issues* to tell people their library isn't adhering to standards.

**Comment by plttn** [comment]
Funnily enough, Octopress 2.0>3.0 did follow semver, as there were major changes. 

In short, 2.0 was distributed by forking that repo which was not an effective way of doing it. 2.0 was not merged into 3.0 as 2.0 is deprecated but still available.

**Comment by pboling** [comment]
TL;DR Don't fucking use Octopress unless you want to read the source and write the migration guide yourself, fuckers!!!!!

**Comment by plttn** [comment]
In the interest of actually answering your question @arielelkin, 2.0 was distributed by forking that repo. @imathis did not believe that was sustainable, so the distribution method was changed (ruby gems), and with that, a new repo was needed.

the one under the octopress org is the one you should be looking at, not the one under imathis's account.

**Comment by jingsam** [comment]
Easy, everyone. Words hurt people. @plttn have clarified the difference, so just relax! @arielelkin thinks that the author should state the difference in documents. However, I do not think it is a responsibility of the author, since he did not charge you anything. @arielelkin should google first before  submit this issue. It is the smart way to ask questions.

**Comment by pboling** [comment]
It is just as much anyone's responsibility to not document (or not) as it is our responsibility to Google.

Though... if it is nowhere documented, what good does Googling do?  That's kind of the point here.

**Comment by holdenweb** [comment]
well, this certainly looks like a project worth avoiding

**Comment by ludwick** [comment]
I just want to remind everyone that:

* The maintainer and everyone who has contributed are volunteers. They have lives and paying jobs they likely prioritize over this. 
* Being hostile or demanding (with or without curse words) causes anyone reading to be less interested in spending their free time on this project or any other.

Please consider how others will interpret your comments and consider that not everyone knows the same things as you.

**Comment by drobban** [comment]
Good answer plttn.
The behaviour of after100 is unacceptable. I understand the reaction of arielelkin.

This will definitely scare away potential contributions and bring bad reputation to the project and the maintainers.

**Comment by tristanfisher** [comment]
I'm not upset and I have no dog in this fight, but I want to express how I wish to be offended by this also!!!!!!

**Comment by apatrida** [comment]
It really isn't clear what the state is, the question was valid and useful.  There is no definitive statement about 3.0 status and all sounds tentative "is coming", "what's coming", "...if you’re curious, you can browse..." but then silence.  And docs I think are meant to be the README.MD from the Octopress repo?

So we see a really old Octopress 2, and a unfinished Octopress 3 with the last news around 11 months old.  No new source commits since July.  Therefore the status is?  Abandoned?  Which is ok, as you said, it is volunteers and if they have lost interest or don't have time, that is good to know as well.  After all, you create a community around your open-source it is respectful of the community to keep them updated even if the news is bad.  

Volunteer or not, you set expectations with your public promotion of your open-source product.  You lure in users.  You market to them with a web page and telling people that your software is good, "use it!" ... so then expect that some non-contributing users will exist, and they will be confused by silence and conditional statements that don't tell them anything definitive.  

Asking for people to contribute now is also not very useful, because even potential contributors would have no idea the state of the project.

Searching Google for Octopress in the past year:

- comparisons against Hexo (Nov 2015), mentions nothing about 3.0
- article about switching to Hugo,... "and I’d been waiting for Octopress 3 for over three years to arrive,..."
- migrating from Octopress 2 to 3, "Octopress is getting ready for a major update to version 3..." and "This article is somewhat premature as there is no release date ('It’s ready when it’s ready!')"
- "Hello Octopress 3" ...  "I was really excited when I the coming of Octopress 3.0 was foretold..."
- "Octopress 3 arrives" ... well no, this article also is about the announcement and saying it is coming.

So nothing says 3.0 is here, everything else says that people are going to other platforms such as Hexo, Hugo, or direct to Jekyll.  

Therefore the status is actually?

**Comment by arielelkin** [comment]
> Volunteer or not, you set expectations with your public promotion of your open-source product. You lure in users. You market to them with a web page and telling people that your software is good, "use it!" ... so then expect that some non-contributing users will exist, and they will be confused by silence and conditional statements that don't tell them anything definitive.

Dead on. 

[I sent out a tweet to octopress three days ago](https://twitter.com/AriVocals/status/666261842566737920) asking for an official clarification. 

No reply. 

If a project owner is neglectful to the extent that they won't bother clarifying, when asked, what the status of their project is, I infer that the owner doesn't care and that their software is not properly maintained. 

The status as I see it? Nobody's home. Even [Hyde](https://en.wikipedia.org/wiki/Strange_Case_of_Dr_Jekyll_and_Mr_Hyde) was proactive.

I'm switching.

**Comment by mjohnq3** [comment]
Read this: [What is Octopress?](https://mjohnq3.github.io/blog/2015/03/31/what-is-octopress/)
Install Ruby on your platform
Install Jekyll (use 2.x for now)
Install Octopress 3.0.11
Read the Octopress README. It's quite useful.
"Blog like a hacker."

**Comment by feriancek** [comment]
Thank you @apatrida for actually answering the question. I am here specificity because I googled looking for the same information @arielelkin was, and ended up here. My google search was literally "when is Octopress 3.0 coming" hoping to catch a blog post about it. This issue was...not what I expected from a mature project.

I'd like to point out that @plttn is empirically incorrect, the gem is listed as version 3.0.11 but is still in an alpha/beta state. It's a re-write so 3.x makes sense, but it should be tagged pre- (or RC if it's that far along). Without a disclaimer saying otherwise, I would've assumed 3.0.0 was a feature complete release, with some bugfixes along the way leading to 3.0.11

But in the end I don't care about the software, there are endless choices for static blog software. You pick one for the community, and this page is telling.

**Comment by parkr** [comment]
> what's the difference between this repo and the one at github.com/imathis/octopress

@arielelkin To answer your question directly, the difference is that this repository contains Octopress 3, and imathis/octopress contains Octopress 2. It's recommended to use Octopress 3 by following the instructions in this repository and using the gem found at `gem install octopress`.
